cli/src/config/merge_tools.toml

```diff
@@ -12,6 +12,20 @@ program="meld"
 # If using this as a template, note that `$output` is repeated twice below
 edit-args = ["$left", "$output", "$right", "-o", "$output"]
 
+[merge-tools.diffedit3]
+program="diffedit3"
+# 17376 is a verified random number, as in https://xkcd.com/221/ :). I am
+# trying to avoid 8000 or 8080 in case those, more commonly used, port
+# numbers are used for something else.
+edit-args = ["$left", "$right", "$output", "--port-range", "17376", "17380"]
```

Comment:
Can `diffedit3` be made to listen on an arbitrary port instead? Maybe by passing `port = 0` by default?

Reply:
I think the answer is no. Port 0 would make the OS pick a random port number, I believe. The app should still print the correct URL in this case , I implemented that but should double-check.

I'm not sure if it's even work to listen on trend of thousands of ports, but if we did, no other application would be able to listen to any of them until we stopped. 

Also, you need sudo to open a port below 1024.

Reply:
> I think the answer is no. Port 0 would make the OS pick a random port number, I believe.

That was my goal :) I think the OS (maybe with the help of the library, somehow) would still not pick the lower 1024 ports.

Reply:
No, it wouldn't.

I didn't want to pick a completely random port to make setting up SSH forwarding easier. I keep meaning to write down that part, but I was going to suggest an SSH config in the docs: port 17376 and maybe 17377, 17378 in the config (or in whatever command-line you usually use to connect to the server), and then `diffedit3` would just work over your connection.

Reply:
I see. Maybe we should make default to using an arbitrary port and add a note about specifying the port range if you're going to use SSH forwarding?

Reply:
Then it'd be quite painful to use SSH forwarding. I don't think we lose much by using the port range, we can make it wider if we need to later (or now; I didn't make it wider mainly to keep the SSH config snippet I was thinking of shorter). Most similar programs I know use a fixed port. In practice, I don't think I ever ran more than 3 instances of `diffedit3` at once.

Another reason: if we always use the same port range, a user can bookmark the URL that's the correct one 90% of the time (unless you run more than one instance of `diffedit3` at once).

In principle, we could add a separate config for `diffedit3-randomport`, but I'm not sure why I'd use it (if I was on a multiuser computer, maybe, where many people used diffedit3? I don't think that'd ever happen in practice, though if it does, we could add the config). If we had a `diffedit3-ssh` config, I'd just recommend everybody use it.

Reply:
I'm not entirely sure it's a good idea, but one thing I could do would be to use port 0 if none of the ports between 17376 and 13480 are open. This would mean that the tool would work for people who don't use SSH, but would be an occasional weird problem that would probably take some debugging instead of an error message for people who do use SSH and encounter this.

I could also rename `diffedit3-nobrowser` to `diffedit3-ssh` and disable this behavior for the latter.

I guess it'd be an option to `diffedit3`, something like `--random-port`. `diffedit3 --help` would explain that this can be used together with `--port` or `--port-range`.

Reply:
I'm currently thinking of adding more complicated syntax for `--port` to `diffedit3` and making the default for `diffedit3` to be `--port 17376-17380 --port 0`, meaning try ports 17376-17380 first, and a random port later. Meanwhile, `diffedit3-ssh` would be `--port 17376-17380 --no-browser`, meaning fail instead of trying a random port and don't launch a browser.

Reply:
Does that sound reasonable to you, @martinvonz ?

Reply:
I don't use SSH forwarding (maybe I should), so I'll just trust you and others on that :)

Reply:
OK, I implemented all this and added the details of how to use SSH forwarding regardless :)